Use global imports for ESPHome

OttoWinter opened the pull request:
## Description:

Looks like integrations can use module-level imports now.

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

OttoWinter committed:
Use global import for ESPHome

balloob commented:
Update `gen_requirements.py` to add aioesphome to the libs to be installed for tests.

OttoWinter committed:
Add aioesphomeapi to test requirements

aioesphomeapi is also shipped as a pure-python wheel, so this should not impact test install time
